packages/shared/lib/core/network/utils/checkNodeUrlValidity.ts

```diff
@@ -19,6 +19,13 @@ export function checkNodeUrlValidity(
         return 'error.node.invalid'
     }
 
+    // Check if it's HTTP and localhost
+    const isHttpLocalhost = newUrl.toLowerCase().startsWith('http://localhost') || newUrl.startsWith('http://127.0.0.1')
```

Comment:
you might do this as well in case the protocol isn't lower case (that would be weird, but you never know)
```suggestion
    const isHttpLocalhost = newUrl.toLowerCase().startsWith('http://localhost') || newUrl.toLowerCase().startsWith('http://127.0.0.1')
```